src/pkg/selectors/onedrive_test.go

```diff
@@ -276,6 +277,7 @@ func (suite *OneDriveSelectorSuite) TestOneDriveCategory_PathValues() {
 	ent := details.DetailsEntry{
 		RepoRef:  filePath.String(),
 		ShortRef: shortRef,
+		ItemRef:  fileID,
```

Comment:
I think for upstream SDK consumers the behavior of `pathValues` for OneDrive/SharePoint will need to be adjusted. We'll need to re-enable matching on item IDs. We should also put out a "known issues" line that states we'll restore/select items with both the name and item ID if users happen to have a file named the same as an item ID. This should be a known issue because we never display OneDrive or SharePoint item IDs to CLI users

Reply:
That will be handled in a later PR.  This change is going to act as a feature branch for two further changes:
1. ensuring population of the item id.
2. selection on item id.

In the second one, I'll allow the selector producer to declare whether it wants to select on item names or item IDs.  The CLI will opt for item names.  Haven't decided if I want to default to including both name and id, or just the id.  The former is more lenient, the latter is reduces some collisions.  Probably going to do the latter.

Either way, once that configuration is in place, we won't have the aforementioned duplicate-id-name collision case.